app/workers/pubsubhubbub/distribution_worker.rb

```diff
@@ -12,10 +12,10 @@ def perform(stream_entry_id)
 
     account = stream_entry.account
     payload = AtomSerializer.render(AtomSerializer.new.feed(account, [stream_entry]))
-    domains = account.followers_domains
+    domains = account.followers_subscription_domains if stream_entry.status&.hidden?
 
     Subscription.where(account: account).active.select('id, callback_url').find_each do |subscription|
-      next unless domains.include?(Addressable::URI.parse(subscription.callback_url).host)
+      next if stream_entry.status&.hidden? && !domains.include?(Addressable::URI.parse(subscription.callback_url).normalize.host)
```

Comment:
Do you want me to use something like:
```ruby
  def is_domain_subscribed?(domain)
    return true if followers.where(domain: domain).count.positive?
    return true if followers.where(web_domain: domain).count.positive?
    followers.reorder(nil).where('accounts.domain IS NOT NULL AND accounts.web_domain IS NULL')
             .select('accounts.remote_url').find_each do |url|
      return true if domain == Addressable::URI.parse(url).normalized_host
    end
    false
  end
```

As I said earlier, I fear this is going to be much slower than what I came up with earlier (at least without the `web_domain` field in the database). But you're right that it should consume less memory.